src/components/archivesSort.tsx

```diff
@@ -0,0 +1,41 @@
+import { Link } from "gatsby"
+import * as React from "react"
+import ShowYear from "../components/showYear"
+
+interface ArchivesSortProps {
+  readonly index: number
+
+  readonly currentNode: {
+    frontmatter: {
+      readonly author: string[]
+      readonly title: string
+      readonly date: string
+    }
+  }
+  readonly preNode: {
+    readonly frontmatter: {
+      readonly date: string
+    }
+  }
+}
+
+export class ArchivesSort extends React.Component<ArchivesSortProps, {}> {
+  render() {
+    const { currentNode, preNode, index } = this.props
+    const title = currentNode.frontmatter.title
+    const date = currentNode.frontmatter.date
+    const author = currentNode.frontmatter.author
+    const preDate = preNode.frontmatter.date
```

Comment:
With [ES6 destructuring](https://hacks.mozilla.org/2015/05/es6-in-depth-destructuring) line 24 can be taken further:

 ```jsx
const { 
  currentNode: {
    frontmatter: {
      title, date, author
    }
  }, 
  preNode, 
  index 
} = this.props
```

Reply:
Wow, Fantastic.